source/_changelogs/3.1.1.md

```diff
@@ -0,0 +1,50 @@
+# 3.1.1
+
+*Released*
+
+**Features:**
+
+- You can now pass a `flag` option to {% url "`cy.writeFile()`" writefile %}. The {% url "following flags" https://nodejs.org/api/fs.html#fs_file_system_flags %} are available.
+- The {% url "Module API" module-api %} now returns the `runUrl` from `cypress.run()`.
+
+**Bugfixes:**
+
+- Fixed a bug where Cypress would error when a secure cookie flag was set and then expired from an `https` url without the subsequent secure flag. Due to the way Cypress was constructing the url, this would fail to actually remove the cookie. Fixes {% issue 1321 %}, {% issue 1264 %} and {% issue 1799 %}.
+- We updated how we handle {% url "`trashAssetsBeforeRuns`" configuration#Screenshots %} behavior. We now trash the contents of the {% url "`screenshotsFolder`" configuration#Folders-Files %} and {% url "`videosFolder`" configuration#Folders-Files %} directories instead of trashing the directories themselves. This helps maintain any file access permissions for the directories. Fixes {% issue 1943 %} and {% issue 2600 %}.
+- Fixed bug where clicking to 'reload' Cypress tests would cause Cypress to freeze with a React error. Fixes {% issue 2590 %}.
+- Fixed bug where mocked Dates using {% url "`cy.clock()`" clock %} would not be an instance of `Date`. Fixes {% issue 2570 %}.
+- Directories within specs whose names contain `.` are now correctly read as directories instead of spec files within the Test Runner. Fixes {% issue 2423 %}.
+- Cypress no longer automatically takes screenshots of skipped tests. Fixes {% issue 2368 %}.
+- Fixed issue where the handler within {% url "`cy.task()`" task %} would incorrectly resolve as `undefined`. Fixes {% issue 2553 %}.
```

Comment:
This should be "Fixed an issue where returning/resolving `undefined` from a cy.task() handler resulted in an incorrect runtime error."